### DISCUSSION TITLE
LAN8742 Support

### DISCUSSION THREAD

**Discussion by carlosmesquita** [arendst/Tasmota#12216] [Ideas]
Hello guys,

Since the LAN8720 is already supported by Tasmota, when using ESP32, and due to the shortage of such chip, would it be possible to implement support for the LAN8742, which is a pin-to-pin compatible alternative, and has (at least for now), some more availability than its predecessor.

From what i've seen, in order to make it work, small changes are needed:
https://github.com/espressif/arduino-esp32/issues/5036
https://github.com/espressif/esp-idf/issues/5092

Basically, the LAN8742 behaves as the LAN8720, but a change in a define is needed, regarding the chip ID.
I can help in developing such feature, i just need some guidance on how this can be performed.

From what i have seen, Tasmota uses ESP-IDF v3? or v4?

Best regards,
Carlos Mesquita

**Comment by Jason2866** [comment]
The needed change is not (yet) made in IDF3.3, or do i have overlooked?
So it is not in the lib which Arduino core uses. Tasmota uses Arduino core not IDF.
We are not going to use a customized Arduino core just to support a LAN chip.

**Reply by carlosmesquita** [reply]
LAN8742 is not yet supported in either Arduino Core or IDF actually...
In order for Tasmota to support it, it must be performed at Arduino Core level, never at Tasmota level, correct?

Ok, so just to confirm, Tasmota uses Arduino Core (latest?), which uses ESP-IDF (currently at v3.3.5, soon to be v4), is that it?

Makes sense, not to use a custom Arduino Core. But if i want to do that myself, can I use a custom Arduino Core "fork" where i implement support for the chip, and maybe try to PR it into Arduino Core/ESP-IDF?

Thanks for your help.

**Reply by sfromis** [reply]
In general, Tasmota does not concern itself with lower level networking code, that is expected to be handled by Arduino Core, which again may delegate functions to ESP-IDF. Tasmota tends to make sure to keep being on compatible and stable versions of those foundations, not necessarily upgrading without a specific reason.

**Reply by carlosmesquita** [reply]
Makes total sense @sfromis , makes it easier to maintain, while also benefiting from newer features that come into Arduino Core.

**Reply by Jason2866** [reply]
Tasmota uses Arduino Core 1.0.6. Since Tasmota is very stable with this core a change to a other core is not planned (there has to be a mandantory reason).
It looks like there is no new updated release for Arduino ESP32 based on IDF 3.3 planned.
If this is the case a change to this core is considered.
The Arduino ESP32 core based on IDF4.4 is in early Alpha build. Tasmota will NOT use this core until it reaches release status. It is used only for the experimental builds for the C3 and S2 (since there is no alternative).
Long story short, LAN8742 chip can not be supported in Tasmota soon.
